trie/triedb/pathdb/history.go

```diff
@@ -576,8 +576,10 @@ func truncateFromHead(db ethdb.Batcher, freezer *rawdb.ResettableFreezer, nhead
 		return 0, err
 	}
 	// Ensure that the truncation target falls within the specified range.
-	if ohead < nhead || nhead < otail {
-		return 0, fmt.Errorf("out of range, tail: %d, head: %d, target: %d", otail, ohead, nhead)
+	if nhead != 0 { // truncating to zero is always possible
```

Comment:
This fix is not correct.

The leftover state history can be in range of [tail, head].
The reset to 0 won't work because the state segment before tail is not longer accessible, but we try to load them here `blobs, err := rawdb.ReadStateHistoryMetaList(freezer, nhead+1, ohead-nhead)`